### PR TITLE
Add support for Kerberos w/ JDBC and HDFS

### PR DESCRIPTION
**Pull request by alexr-bq**
### Summary

Add support for Kerberos environment.

### Description

This handles two pieces of functionality:
- Passing Kerberos parameters to JDBC
- If Kerberos is enabled, pass a Kerberos impersonation token for HDFS to Vertica

Testing environment used for this has a PR here: https://github.com/jonathanl-bq/vertica-testenv/pull/1

### Related Issue

VER-76086
AND
VER-76703

### Additional Reviewers

@jonathanl-bq 
@NerdLogic

### CODE REVIEW

**Review by jonathanl-bq — APPROVED** [review]

File: examples/kerberos-example/output.txt [hunk 1]
```diff
@@ -0,0 +1,65 @@
+[info] welcome to sbt 1.4.1 (N/A Java 14.0.1)
```
Comment (alexey-temnikov) [inline examples/kerberos-example/output.txt:1]: This is auto generated file, should not be part of the repo?

File: connector/src/main/scala/com/vertica/spark/config/JDBCConfig.scala [hunk 2]
```diff
@@ -15,14 +15,46 @@ package com.vertica.spark.config
 
 import ch.qos.logback.classic.Level
 
+/**
+ * Represents any config necessary for authenticating to JDBC.
+ *
```
Comment (alexey-temnikov) [inline connector/src/main/scala/com/vertica/spark/config/JDBCConfig.scala:20]: .nit extra line.

File: examples/kerberos-example/output.txt (hunk 1 above)
Comment (alexr-bq) [reply examples/kerberos-example/output.txt:1]: Yep, will remove

File: connector/src/main/scala/com/vertica/spark/config/JDBCConfig.scala (hunk 2 above)
Comment (alexr-bq) [reply connector/src/main/scala/com/vertica/spark/config/JDBCConfig.scala:20]: This is expected scaladoc style afaik

**Review by alexey-temnikov — APPROVED** [review]

**Pull request merged**